cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaMessageCallback.java

```
@@ -94,7 +95,7 @@ public void onReceived(Iterator<FetchedMessage> messages) {
       } catch (Throwable th) {
         LOG.error("Error processing message at topic {} parition {}",
```

Since the exception is not propagated, I think it's good to indicated that in the error message as well, something like "Message ignored due to exception for topic {}, partition {}, message {}"...

---

If logging the message is too large, it would be good to log the message offset so that it can be debug later (as long as it's still in Kafka).